### ISSUE TITLE
Allow to set outfilesuffix for a Java converter

### ISSUE THREAD

**Issue by robertpanzer** [asciidoctor/asciidoctorj#354]
Currently Java converters always create files with the extension .html

It should be possible to change this.

Options are:
- Allow to define the suffix via an annotation. If so, add another parameter to the `@ConverterFor` annotation? (Would not read so nicely anymore, change the name to sth else?)
- Add a method `Converter.setOutfilesuffix()` that must be called in the constructor of the converter if it should change.

**Comment by mojavelinux** [comment]
You know I like the annotations :)

I think the `@ConverterFor` should accept an `extension`, `suffix` or similar attribute:

```
@ConverterFor("markdown", suffix = ".md")
```

I'm okay with adding the method for a programmatic setting, but the annotation should be the preferred approach.

I think the method should be: `setOutfileSuffix` to be consistent with Java naming.

**Comment by robertpanzer** [comment]
Sounds good!

But I think with two parameters the annotaton will look like this:
```
@ConverterFor(value = "markdown", suffix = ".md")
```

**Comment by robertpanzer** [comment]
Maybe better to add another parameter `target` so that we can write
```
@ConverterFor(target= "markdown", suffix = ".md")
```

With one parameter this could still be 
```
@ConverterFor("foo")
```
if `.html` should be the file extension.

**Comment by mojavelinux** [comment]
Ah, you are correct. I'm okay with the `value` attribute. I think `target` introduces a new term that isn't well defined here.

**Comment by robertpanzer** [comment]
We used the term _target format_ in the README.
Maybe `@ConverterFor(targetFormat="abc", suffix=".md")`?

**Comment by mojavelinux** [comment]
In that case, I might go with `format`. I'm not sure "target" by itself means anything in this context without "format". It's just a clarifying adjective.

**Comment by mojavelinux** [comment]
```
@ConverterFor(target="markdown", suffix=".md")
```

reads very nicely

**Comment by mojavelinux** [comment]
Oops, ignore what you just saw and see what is there now :) I made a typo.

**Comment by robertpanzer** [comment]
:) Happens to me also very often

**Comment by robertpanzer** [comment]
Fixed with #355
